pkg/retistruct/R/server-handlers.R

```diff
@@ -276,6 +286,17 @@ h.save <- function(h, state, ...) {
 
 ## Handler for reconstruction
 h.reconstruct <- function(h, state, input, output, session, ...) {
+  validate(
+    need(is.numeric(input$center.el), "Center el is not a numeric. Check that the input is not an expression, or non-empty."),
+    need(is.numeric(input$center.az), "Center az is not a numeric. Check that the input is not an expression, or non-empty."),
+    need(is.numeric(input$ax.el), "Axis el is not a numeric. Check that the input is not an expression, or non-empty."),
+    need(is.numeric(input$ax.az), "Axis az is not a numeric. Check that the input is not an expression, or non-empty."),
+    need(0 <= input$center.el && input$center.el <= 90, "Center el must be between 0 and 90"),
+    need(0 <= input$center.az && input$center.az <= 90, "Center az must be between 0 and 90"),
+    need(0 <= input$ax.el && input$ax.el <= 90, "Ax el must be between 0 and 90"),
+    need(0 <= input$ax.az && input$ax.az <= 90, "Ax az must be between 0 and 90")
+  )
+
```

Comment:
See comment from above:

Thanks for making this change. However, for both the centre of the projection and the rotation axis, the elevation (el) should be between -90 and 90 inclusive and the azimuth (az) should be in the range -180 to 180.

Also, could we rename "center" to "centre" - Retistruct uses British English :)

I've put in another suggestion for an enhancement, which I propose we *don't* do for v0.8.0: #64